core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java

```diff
@@ -86,4 +87,18 @@ public static boolean contains(Object[] array, Object element) {
 
 		return result;
 	}
+
+	public static <T> T[] concat(List<T[]> arrays) {
+		T[] result = arrays.get( 0 );
```

Comment:
Hum, I think I'd allocate the result once in the final size and then populate it array by array.